zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -931,47 +939,54 @@ pub(crate) fn add_transparent_account_balances(
             .with_unshielded_balance_mut(|bal| bal.add_spendable_value(value))?;
     }
 
-    let mut stmt_account_unconfirmed_balances = conn.prepare(
-        "SELECT a.uuid, SUM(u.value_zat)
-         FROM transparent_received_outputs u
-         JOIN accounts a ON a.id = u.account_id
-         JOIN transactions t ON t.id_tx = u.transaction_id
-         -- the transaction that created the output is mined with not enough confirmations or is definitely unexpired
-         WHERE (
-            t.mined_height < :mempool_height
-            AND :mempool_height - t.mined_height < :min_confirmations -- tx is mined but not confirmed
-            OR t.expiry_height = 0 -- tx will not expire
-            OR t.expiry_height >= :mempool_height
-         )
-         -- and the received txo is unspent
-         AND u.id NOT IN (
-           SELECT transparent_received_output_id
-           FROM transparent_received_output_spends txo_spends
-           JOIN transactions tx
-             ON tx.id_tx = txo_spends.transaction_id
-           WHERE tx.mined_height IS NOT NULL -- the spending tx is mined
-           OR tx.expiry_height = 0 -- the spending tx will not expire
-           OR tx.expiry_height >= :mempool_height -- the spending tx is unexpired
-         )
-         GROUP BY a.uuid",
-    )?;
-
-    let mut rows = stmt_account_unconfirmed_balances.query(named_params![
-        ":mempool_height": u32::from(mempool_height),
-        ":min_confirmations": min_confirmations,
-    ])?;
+    // Pending spendable balance for transparent UTXOs is only relevant for min_confirmations > 0;
+    // with min_confirmations == 0, zero-conf spends are allowed and therefore the value will
+    // appear in the spendable balance and we don't want to double-count it.
+    if min_confirmations > 0 {
+        let mut stmt_account_unconfirmed_balances = conn.prepare(
+            "SELECT a.uuid, SUM(u.value_zat)
+             FROM transparent_received_outputs u
+             JOIN accounts a ON a.id = u.account_id
+             JOIN transactions t ON t.id_tx = u.transaction_id
+             WHERE (
+                 -- the transaction that created the output is mined with not enough confirmations
+                (
+                    t.mined_height < :mempool_height
+                    AND :mempool_height - t.mined_height < :min_confirmations -- tx is mined but not confirmed
+                )
+                -- or the tx is unmined but definitely not expired
+                OR (t.expiry_height = 0 OR t.expiry_height >= :mempool_height)
```

Comment:
Bug: This is not an unmined check. We can reach this conditional in two cases:
- `t.mined_height >= :mempool_height` (TXO is unmined - good!)
- `:mempool_height - t.mined_height >= :min_confirmations` (TXO is mined and confirmed - bad!)

I believe this is an existing bug, because `AND` has higher precedence than `OR`, so the previous query had the same SQLite operator precedence as this query does after parens insertion. However it is a transient bug because the mined TXO stops being counted once it would have expired. This bug and its transient-ness should be testable.